tests/rust/swift_name.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::Arc;
+
+struct Foo {
+  bar: i32,
+}
+
+#[repr(C)]
+struct FooRef {
+  ptr: *mut Foo
+}
+
+impl FooRef {
+  #[export_name="FooRef_create"]
+  pub extern fn create() -> FooRef {
+    Box::into_raw(Box::new(Foo {
+      bar: 0
+    }))
+  }
+
+  #[export_name="FooRef_setBar"]
+  pub extern fn set_bar(self: FooRef, bar: i32) {
+    if let Some(nonnull) = std::ptr::NonNull::new(self.ptr) {
+      nonnull.as_mut().bar = bar;
+    }
+  }
+
+  #[export_name="FooRef_getBar"]
+  pub extern fn get_bar(self: FooRef) -> i32 {
```

Comment:
So what happens when you use `&self` / `&mut self` / etc? What about `self: Box<Self>` or such?

Can we have tests for that? We should at least not generate broken code.